packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -62,6 +62,11 @@ export interface IFunction extends IResource, ec2.IConnectable, iam.IGrantable {
    */
   readonly architecture: Architecture;
 
+  /**
+   * The ARN(s) to put into the resource field of the generated IAM policy for grantInvoke()
+   */
+  readonly resourceArnsForGrantInvoke: string[];
+
```

Comment:
Why are you adding this to the interface? (Not saying it is incorrect, just want to understand why)

Reply:
Most/all integrations accept an `IFunction`, so we don't have access to `FunctionBase.resourceArnsForGrantInvoke` and instead need to use `IFunction.resourceArnsForGrantInvoke`. This is true for basically all of the stepfunctions-tasks lambda integrations.